packages/web-components/src/slider/slider.ts

```diff
@@ -0,0 +1,26 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTSlider } from '@microsoft/fast-foundation';
+import type { SliderSize } from './slider.options.js';
+
+/**
+ * The base class used for constructing a fluent-slider custom element
+ * @public
+ */
+export class Slider extends FASTSlider {
+  /**
+   * The size of the slider
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   */
+  @attr
+  public size: SliderSize = 'medium';
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    if (this.step) {
+      const totalSteps = 100 / Math.floor((this.max - this.min) / this.step);
+      this.style.setProperty('--step-rate', totalSteps.toString() + '%');
+    }
```

Comment:
This is only going to be set once, should this be in the `stepChanged` function? You can confirm the behavior by initializing without steps and then setting the attribute...or changing an existing step attribute. Would we also want to _remove_ this property in the case that step was removed?

Reply:
I might not have enough knowledge here, and I couldn't find an example or text in the docs to answer this question: If I put this in the stepChanged() function how can I set the initial value when the component is initially rendered? Is stepChanged() only called when the attribute is changed post render?

Reply:
@procload my understanding was to not show steps by default - has that changed?

Reply:
That's correct. But when I moved the logic into the stepChanged() function, it no longer detects the step attribute until I manually change that value in the DOM via JavaScript (or the Inspector).

Reply:
I'll pull your branch down...to confirm, when you have the below code it's not firing on connect?

```hml
<fluent-slider step="5"></fluent-slider>
```

Reply:
@procload 

```ts
private stepChanged(): void {
    super.stepChanged();

    if (this.step) {
      const totalSteps = 100 / Math.floor((this.max - this.min) / this.step);

      this.style.setProperty('--step-rate', totalSteps.toString() + '%');
    }
  }
```
I see the property being set on the host element. I don't see the styles applying like you might expect, but it's there with the above from what I can see.

Reply:
Probably want to check the math on the above...I'm getting the following in the console for all stories when I log the value of `totalSteps`:
<img width="1218" alt="image" src="https://user-images.githubusercontent.com/13071055/225469533-fcdde7a9-3234-452e-bc7e-b2f1544f3560.png">